packages/sources/dxfeed-test/src/endpoint/price-ws.ts

```diff
@@ -0,0 +1,146 @@
+import { WebSocketTransport } from '@chainlink/external-adapter-framework/transports/websocket'
+import { makeLogger, SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { WebSocket } from '@chainlink/external-adapter-framework/transports'
+import { customSettings } from '../config'
+
+const logger = makeLogger('DxFeed Websocket')
+
+export type DXFeedMessage = {
+  channel: string
+  clientId?: string
+  id: string
+  data: [string, [string, number, number, number, number, string, number, string, number, number]]
+  successful?: boolean
+  advice?: {
+    interval: number
+    timeout: number
+    reconnect: string
+  }
+}[]
+
+export type EndpointTypes = {
+  Request: {
+    Params: { base: string }
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: DXFeedMessage
+  }
+}
+
+const META_HANDSHAKE = '/meta/handshake'
+const META_CONNECT = '/meta/connect'
+const SERVICE_SUB = '/service/sub'
+const SERVICE_DATA = '/service/data'
+
+const handshakeMsg = [
+  {
+    id: 1,
+    version: '1.0',
+    minimumVersion: '1.0',
+    channel: META_HANDSHAKE,
+    supportedConnectionTypes: ['websocket', 'long-polling', 'callback-polling'],
+    advice: {
+      timeout: 60000,
+      interval: 0,
+    },
+  },
+]
+
+const firstHeartbeatMsg = [
+  {
+    id: 2,
+    channel: META_CONNECT,
+    connectionType: 'websocket',
+    clientId: '',
+    advice: {
+      timeout: 60000,
+    },
+  },
+]
+
+const heartbeatMsg = [
+  {
+    id: 3,
+    channel: META_CONNECT,
+    clientId: '',
+    connectionType: 'websocket',
+  },
+]
+
+class DxFeedWebsocketTransport extends WebSocketTransport<EndpointTypes> {
+  connectionClientId = ''
```

Comment:
Instead of doing the messages as consts above, can we move them inside of this class and have them just get the `connectionClientId` from within the class? That way we avoid having to remember to set it in the messages before sending